File: codeforces/1800~1899/1860/a.cc
#include <bits/stdc++.h>

using namespace std;

using i64 = long long;

void solve() {
  string s;
  cin >> s;

  if (s == "()") {
    cout << "NO" << endl;
    return;
  }

  int n = s.length();
  bool found = false;
  for (int i = 1; i < n; ++i) {
    if (s[i] == s[i-1]) {
      found = true;
      break;
    }
  }

  string ans;
  if (found) {
    for (int i = 0; i < n; ++i) {
      ans += "()";
    }
  } else {
    ans = string(n, '(') + string(n, ')');
  }
  cout << "YES" << endl;
  cout << ans << endl;
}

int main(void) {
  ios::sync_with_stdio(false);
  cin.tie(0);

  int T = 0;
  cin >> T;
  for (int t = 0; t < T; ++t) {
    solve();
  }
  return 0;
}

File: codeforces/1800~1899/1860/a.rs
use std::io;
use std::str::FromStr;
use std::fmt::Debug;

fn read_int<T: FromStr>() -> T where <T as FromStr>::Err: Debug {
    let mut s = String::new();
    io::stdin().read_line(&mut s).unwrap();
    s.trim().parse::<T>().expect("invalid input")
}

fn solve() {
    let mut s = String::new();
    io::stdin().read_line(&mut s).unwrap();
    s = s.trim().to_string();

    if s == "()" {
        println!("NO");
        return;
    }

    let n = s.len();
    let mut found = false;
    for i in 1..n {
        if s.chars().nth(i) == s.chars().nth(i-1) {
            found = true;
            break;
        }
    }

    let mut ans = String::new();
    if found {
        for _i in 0..n {
            ans += "()";
        }
    } else {
        ans = "(".repeat(n) + &")".repeat(n);
    }

    println!("YES");
    println!("{}", ans);
}

fn main() {
    let n = read_int::<i32>();
    for _t in 0..n {
        solve();
    }
}
